### PR TITLE
[new release] pp (1.1.2)

### PR DESCRIPTION
**Pull request by jeremiedimino**
Pretty-printing library

- Project page: <a href="https://github.com/ocaml-dune/pp">https://github.com/ocaml-dune/pp</a>
- Documentation: <a href="https://ocaml-dune.github.io/pp/">https://ocaml-dune.github.io/pp/</a>

##### CHANGES:

- Add `of_fmt` to compose with existing pretty printers written in `Format`
  (ocaml-dune/pp#1, @Drup).

- Use a tail-recursive `List.map` to fix a stack overflow issue (ocaml-dune/pp#3,
  @emillon)

- Add `Pp.custom_break` (ocaml-dune/pp#4, @gpetiot)

- Add `Ast` sub-module to expose a stable representation for
  serialization, allowing to do the rendering in another process (ocaml-dune/pp#6,
  @rgrinberg)

### CODE REVIEW

**Comment by mseri** [comment]
There are some failures on
```
#=== ERROR while compiling pp.1.1.2 ===========================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/ocaml-dune/pp/releases/download/1.1.2/pp-1.1.2.tbz)
# path                 ~/.opam/4.07/.opam-switch/build/pp.1.1.2
# command              ~/.opam/4.07/bin/dune build -p pp -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/pp-15-89b24d.env
# output-file          ~/.opam/log/pp-15-89b24d.out
### output ###
#       ocamlc src/.pp.objs/byte/pp.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.pp.objs/byte -intf-suffix .ml -no-alias-deps -o src/.pp.objs/byte/pp.cmo -c -impl src/pp.ml)
# File "src/pp.ml", line 151, characters 30-51:
# Error: Unbound value pp_print_custom_break
#     ocamlopt src/.pp.objs/native/pp.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlopt.opt -w -40 -g -I src/.pp.objs/byte -I src/.pp.objs/native -intf-suffix .ml -no-alias-deps -o src/.pp.objs/native/pp.cmx -c -impl src/pp.ml)
# File "src/pp.ml", line 151, characters 30-51:
# Error: Unbound value pp_print_custom_break
```
but I don't know what bounds I need to add for it

File: packages/pp/pp.1.1.2/opam [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+homepage: "https://github.com/ocaml-dune/pp"
+doc: "https://ocaml-dune.github.io/pp/"
+bug-reports: "https://github.com/ocaml-dune/pp/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04.0"}
+  "ppx_expect" {with-test}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (mseri) [inline packages/pp/pp.1.1.2/opam:36]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by jeremiedimino** [comment]
Seems like we need at least 4.08. @rgrinberg is 4.08 fine for your use case?

**Comment by rgrinberg** [comment]
Yes, that's the lower bound I use everywhere as well.

File: packages/pp/pp.1.1.2/opam [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  "Jane Street Group, LLC <opensource@janestreet.com>"
+  "Jeremie Dimino <jeremie@dimino.org>"
+]
+license: "MIT"
+homepage: "https://github.com/ocaml-dune/pp"
+doc: "https://ocaml-dune.github.io/pp/"
+bug-reports: "https://github.com/ocaml-dune/pp/issues"
+depends: [
+  "dune" {>= "2.0"}
+  "ocaml" {>= "4.04.0"}
```
Comment (mseri) [inline packages/pp/pp.1.1.2/opam:32]:
```suggestion
  "ocaml" {>= "4.08.0"}
```

Let's try then!

**Comment by jeremiedimino** [comment]
Ack, I updated the upstream opam file as well

**Comment by kit-ty-kate** [comment]
Looks good now. Thanks!

**Pull request merged**